web/src/views/DandisetLandingView/DandisetPublish.vue

```diff
@@ -305,11 +305,15 @@ const alreadyBeingPublishedError = ref(false);
 const containsZarr = ref(false);
 watchEffect(async () => {
   if (currentDandiset.value) {
-    const zarr = await dandiRest.zarr({
-      dandiset: currentDandiset.value.dandiset.identifier,
-    });
-
-    containsZarr.value = zarr.count > 0;
+    const { identifier } = currentDandiset.value.dandiset;
+    const { version } = currentDandiset.value;
+    const res = await dandiRest.assets(
+      identifier,
+      version,
+      { params: { zarr: true, page_size: 1 } },
+    );
+
+    containsZarr.value = res.count > 0;
```

Comment:
Our `asset()` function returns `null` in case of a 404 error, so I think the following adjustment will solve the typechecking error.

```suggestion
    containsZarr.value = res !== null && res.count > 0;
```